pkg/templates/antiaffinity/template.go

```diff
@@ -59,30 +61,48 @@ func init() {
 				if !hasPods {
 					return nil
 				}
-				if affinity := podTemplateSpec.Spec.Affinity; affinity != nil && affinity.PodAntiAffinity != nil {
-					preferredAffinity := affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-					requiredAffinity := affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution
-					for _, preferred := range preferredAffinity {
-						if affinityTermMatchesLabelsAgainstNodes(preferred.PodAffinityTerm, podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher) {
-							return nil
-						}
+				affinity := podTemplateSpec.Spec.Affinity
+				// Short-circuit if no affinity rule is specified within the pod spec.
+				if affinity == nil || affinity.PodAntiAffinity == nil {
+					return []diagnostic.Diagnostic{
+						{Message: fmt.Sprintf("object has %d %s but does not specify inter pod anti-affinity",
+							replicas, stringutils.Ternary(replicas > 1, "replicas", "replica"))},
 					}
-					for _, required := range requiredAffinity {
-						if affinityTermMatchesLabelsAgainstNodes(required, podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher) {
-							return nil
-						}
+				}
+				var foundIssues []diagnostic.Diagnostic
+				preferredAffinity := affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
+				requiredAffinity := affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+				for _, preferred := range preferredAffinity {
+					err := affinityTermMatchesLabelsAgainstNodes(preferred.PodAffinityTerm,
+						podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher)
+					if err != nil {
+						foundIssues = append(foundIssues, diagnostic.Diagnostic{
+							Message: err.Error(),
+						})
+					}
+				}
+				for _, required := range requiredAffinity {
+					err := affinityTermMatchesLabelsAgainstNodes(required, podTemplateSpec.Namespace,
+						podTemplateSpec.Labels, topologyKeyMatcher)
+					if err != nil {
+						foundIssues = append(foundIssues, diagnostic.Diagnostic{
+							Message: err.Error(),
+						})
 					}
 				}
-				return []diagnostic.Diagnostic{
-					{Message: fmt.Sprintf("object has %d %s but does not specify inter pod anti-affinity", replicas, stringutils.Ternary(replicas > 1, "replicas", "replica"))},
+				if foundIssues != nil {
+					return foundIssues
 				}
+				return nil
 			}, nil
 		}),
 	})
 }
 
-func affinityTermMatchesLabelsAgainstNodes(affinityTerm coreV1.PodAffinityTerm, podNamespace string, podLabels map[string]string, topologyKeyMatcher func(string) bool) bool {
-	// If namespaces is not specified in the affinity term, that means the affinity term implicitly applies to the pod's namespace.
+func affinityTermMatchesLabelsAgainstNodes(affinityTerm coreV1.PodAffinityTerm, podNamespace string,
```

Comment:
Nit: this function should be renamed now that it's no longer returning a `bool`. Perhaps `validateAffinityTermMatchesAgainstNodes`?